be/src/vec/olap/olap_data_convertor.h

```diff
@@ -182,12 +182,56 @@ class OlapBlockDataConvertor {
         Status convert_to_olap() override;
     };
 
+    // For compatibility with decimalv2.
+    // Convert the decimal32/decimal64/decimal128 of the computing
+    // layer to decimal12_t of the storage layer.
+    template <typename T>
     class OlapColumnDataConvertorDecimal
             : public OlapColumnDataConvertorPaddedPODArray<decimal12_t> {
     public:
         Status convert_to_olap() override;
     };
 
+    // decimalv3 don't need to do any convert
+    template <typename T>
+    class OlapColumnDataConvertorDecimalV3 : public OlapColumnDataConvertorBase {
```

Comment:
should be a subclass of `OlapColumnDataConvertorSimple` only need override the `convert_to_olap` method